core/block.js

```diff
@@ -881,6 +882,37 @@ Blockly.Block.prototype.getColourTertiary = function() {
   return this.colourTertiary_;
 };
 
+/**
+ * Get the shadow colour of a block.
+ * @return {?string} #RRGGBB string.
+ */
+Blockly.Block.prototype.getColourShadow = function() {
+  var colourSecondary = this.getColourSecondary();
+  if (colourSecondary) {
+    return colourSecondary;
+  }
+  var rgb = goog.color.hexToRgb(this.getColour());
+  rgb = goog.color.lighten(rgb, 0.6);
+  return goog.color.rgbArrayToHex(rgb);
+};
+
+Blockly.Block.prototype.getColourBorder = function() {
```

Comment:
It returns an object because if the block isn't using a style the block sort of has two borders, a light one and a dark one. Another option would be to return either the tertiary colour, or an object containing the light and dark colors, but I thought having a consistent return type might be simpler.

Reply:
Okay.  I think that makes sense, but let's mark this function package rather than public.  Then I'll hit merge.